client/src/types/DictionaryType.d.ts

```diff
@@ -0,0 +1,5 @@
+type DictionaryType = {
+    [k:string]:{
+        [l:string]: string
+    }
+};
```

Comment:
I dont really get it, but looks beautiful.

Reply:
This type describe an object, where the 'k' key is string ( we do NOT specify the NUMBER of 'k' keys! )
K's value is another object, where the key 'l' is a string, and l's value is also a string ( we do not specify the NUMBER of 'l' keys!)

The reason:
We do not know at all how many 'k' and 'l' keys do we have in the language dictionaries!
The specification above describe a GENERIC form of the dictionaries, w/o the need of specify the number of 'k'-s or 'l'-s!

This means, this is a valid Dict. type:
{
"transactionTypes": {
        "PAYMENT": "payment",
        "DEBT": "debt"
    },

    "addTransactionDialog": {
        "BEFORETYPE": "Add ",
        "AFTERTYPE": "",
        "DATESELECTORINFO": "click on the field to select date"
    }
}
Here we have 2x'k' and 2x'l' and 3x'l'. So extend the Dictinary is easy.